source/faq.txt

```diff
@@ -19,7 +19,87 @@ This page contains frequently asked questions and their corresponding answers.
    If you can't find an answer to your problem on this page,
    see the :ref:`golang-issues-and-help` page for next steps and more
    resources.
+
+.. _golang-faq-connection-pool:
+
+How Does Connection Pooling Work in the {+driver-short+}?
+---------------------------------------------------------
+
+Every ``Client`` instance has a built-in connection pool for each server
+in your MongoDB topology. These pools open sockets on demand to support
+concurrent MongoDB operations, or `goroutines
+<https://www.golang-book.com/books/intro/10>`__, in your application.
+
+The maximum size of each connection pool is set by ``maxPoolSize``, which
+defaults to ``100``. If the number of connections to a server is
+the value of ``maxPoolSize`` and every connection is in use, the next
+request to that server will wait until one of the connections becomes
+available.
```

Comment:
It took me a few readthroughs to fully grasp this sentence. I think it could be simplified a bit. Maybe something like:

```suggestion
defaults to ``100``. If the number of in-use connections to a server reaches
the value of ``maxPoolSize``, the next
request to that server will wait until one of those connections becomes available.
```